packages/venia-ui/lib/components/Checkout/cart.js

```diff
@@ -1,19 +1,152 @@
-import React from 'react';
-import { bool, func, shape, string } from 'prop-types';
+import React, { useEffect } from 'react';
+import { func, shape, string } from 'prop-types';
 
 import { mergeClasses } from '../../classify';
 import CheckoutButton from './checkoutButton';
 import defaultClasses from './cart.css';
+import { useRestApi } from '@magento/peregrine';
+// import { useUserContext } from '@magento/peregrine/lib/state/User';
+import { useCheckoutContext } from '@magento/peregrine/lib/state/Checkout';
+import { useCartContext } from '@magento/peregrine/lib/state/Cart';
+import { useDirectoryContext } from '@magento/peregrine/lib/state/Directory';
 
+const AUTHED_SHIPPING_METHOD_ENDPOINT =
+    '/rest/V1/carts/mine/estimate-shipping-methods';
+const AUTHED_CART_ENDPOINT = '/rest/V1/carts/mine';
+const COUNTRIES_ENDPOINT = '/rest/V1/directory/countries';
+const GUEST_CART_ENDPOINT = '/rest/V1/guest-carts';
+
+const SHIPPING_METHOD_OPTIONS = {
+    method: 'POST',
+    body: JSON.stringify({
+        address: {
+            country_id: 'US',
+            postcode: null
+        }
+    })
+};
+
+const CART_METHOD_OPTIONS = {
+    method: 'POST'
+};
+
+const isCartReady = cart => cart.details && cart.details.items_count > 0;
+
+const isCheckoutReady = (cartState, checkoutState, directoryState) => {
+    const hasCountries = !!directoryState.countries;
+    const hasShippingMethods = checkoutState.availableShippingMethods.length;
+    return hasCountries && hasShippingMethods && isCartReady(cartState);
+};
 const Cart = props => {
-    const { beginCheckout, ready, submitting } = props;
+    const { cart: cartState, user: userState } = props;
+    // TODO: Use new state when we migrate cart and user over.
```

Comment:
Not entirely. The state and some initial set/update reducers have been written but the code that creates/queries cart info is not created. This is using the connected redux cart for now. One way around this would be to have the redux actions update the new cart context as well as the redux context. Then we could move each action over separately.